c/src/sbp.c

```diff
@@ -381,8 +457,15 @@ s8 sbp_process(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void *context))
       crc = crc16_ccitt(&(s->msg_len), 1, crc);
       crc = crc16_ccitt(s->msg_buff, s->msg_len, crc);
       if (s->crc == crc) {
-
-        /* Message complete, process it. */
+        s->frame_buff[SBP_HEADER_LEN + s->msg_len] = crc & 0xFF;
+        s->frame_buff[SBP_HEADER_LEN + s->msg_len + 1] = (crc >> 8) & 0xFF;
+        /* Message complete, process frame callbacks, then msg_type callbacks.
+         * Note, return code is ignored from sbp_process_frame.
+         * In future, it may be wise to add new return codes for the truth
+         * table of both process function return codes. For now, we return only
+         * sbp_process_payload's return code for backwards compatibility. */
+        sbp_process_frame(s, s->sender_id, s->msg_type, s->msg_len, s->msg_buff,
```

Comment:
I would propose we allow `sbp_process_frame` to take an argument such that it calls all or a subset of callback types. this would allow `sbp_process_payload` to be a specialized call to `sbp_process_frame` where only `SBP_TYPE_CALLBACK` nodes are executed (similar to how you are extending the callback register functions

Reply:
this would reduce code duplication between the two methods as well as avoid having to traverse the list twice.